base/compiler/tfuncs.jl

```diff
@@ -876,12 +876,12 @@ function _fieldtype_nothrow(@nospecialize(s), exact::Bool, name::Const)
     isa(fld, Int) || return false
     ftypes = datatype_fieldtypes(u)
     nf = length(ftypes)
-    (fld >= 1 && fld <= nf) || return false
     if u.name === Tuple.name && fld >= nf && isvarargtype(ftypes[nf])
-        # The length of the tuple will be determined at runtime, we can't say
-        # anything
-        return false
+        # If we don't know the exact type, the length of the tuple will be determined
+        # at runtime and we can't say anything.
+        return exact
```

Comment:
This still doesn't seem right. `fieldtype(NTuple{10, T} where T, 20)` certainly does error.

Reply:
That doesn't contain a `Vararg` but I guess it might in the future, so I can add it.

Reply:
Fair enough. I guess we didn't change this yet.